### PR TITLE
Pick ClickHouse, Presto and Exasol dialect from Calcite

### PR DESCRIPTION
**Pull request by Leon-WTF**

### CODE REVIEW

**Comment by cgivre** [comment]
Just a thought here, but would it make sense to add additional dialects such as Presto as well?

File: ubenchmark/pom.xml [hunk 1]
```diff
@@ -20,7 +20,7 @@ limitations under the License.
   <parent>
     <groupId>org.apache.calcite</groupId>
     <artifactId>calcite</artifactId>
-    <version>1.21.0-drill-r3</version>
+    <version>1.21.0-drill-r4</version>
```
Comment (vdiravka) [inline ubenchmark/pom.xml]: We might want to consider flatten maven plugin to keep the version in one place. But this is for other PR.
_upd:_ in future Calcite versions Gradle is used. So possibly this issue will gone

**Review by vdiravka — COMMENTED** [review]
> Just a thought here, but would it make sense to add additional dialects such as Presto as well?

Good idea. That should be two additional cherry-picks:
* CALCITE-3724 for for `PRESTO`
* CALCITE-4614 for `EXASOL`

@Leon-WTF If any conflicts will arise, please don't proceed and leave it for future Calcite update version, because that is not related to your CH update.

**Comment by vdiravka** [comment]
@Leon-WTF 
Could you please use `git cherry-pick` to track original commit annotation metadata?
Why do you remove the commit with bump-up Calcite version?

**Comment by Leon-WTF** [comment]
> @Leon-WTF
> Could you please use `git cherry-pick` to track original commit annotation metadata?
> Why do you remove the commit with bump-up Calcite version?

I thought commit with bump-up Calcite version should be the last commit, I added the commits for picking up the other two dialects and then redo commit with bump-up Calcite version.
I done it with patch, will try with  git cherry-pick

**Comment by Leon-WTF** [comment]
> @Leon-WTF
> Could you please use `git cherry-pick` to track original commit annotation metadata?
> Why do you remove the commit with bump-up Calcite version?

@vdiravka Could you help to reopen this PR? It closed automaticly when I want to rollback all commits and redo with cherry-pick. Thanks

**Comment by Leon-WTF** [comment]
> @Leon-WTF
> Could you please use `git cherry-pick` to track original commit annotation metadata?
> Why do you remove the commit with bump-up Calcite version?

@vdiravka I have redo with `git cherry-pick`, could you help to reopen this PR and review? Thanks

**Comment by vdiravka** [comment]
Looks like I haven't possibility to perform that. You can try: https://github.com/gitbucket/gitbucket/issues/378
But feel free to open a new one.

And please name the feature branch with a ticket number and name PR with a ticket number and ticket name. 
For example: https://github.com/vvysotskyi/drill-calcite/pull/3

**Pull request closed without merging**